### DISCUSSION TITLE
Environmental Effect - Water volumes

### DISCUSSION THREAD

**Discussion by AMZN-Liv** [o3de/rfcs#24] [Ideas]
**Is your feature related to a problem? Please describe**
 The current process for creating water effects in O3DE is to rely on animated meshes or use shaders. Without a built-in water authoring tool, it is challenging to create realistic water effects that interact with other objects naturally. 

**Describe the solution you'd like**
 A centralized system, perhaps as part of the terrain editor, for authoring volumetric water effects in-editor. Ideally, this solution would also support infinite ocean water effects as well.

**Comment by HogJonny-AMZN** [comment]
If we compare to legacy Lumberyard, there was a Water Gem.
Water is actually a number of features, tools and possibly more then one workflow.
The feature components would be something like:
- water material/shaders (surface rendering, below surface refraction, particulates and light extinction, foam, shore waves, etc.)
- water surfaces / meshes:
- surface from volume, requires using shapes or volume components and placed in a level
- river surfaces, these are a combination of a spline component, ribbon (mesh) component, directional UV mapping (to bend material), etc.
- underwater rendering includes things like:
- volumetrics (like light shafts)
- PostFX for:
- distortion
- caustics
- etc.

The entity/component system design promotes the re-use of modular components, so for example most things that are defined by a volume use some variation of box or other existing shape components.

Likely some portions of rendering can be shared, for example underwater volumetric lighting (like light shafts) may possibly share some generalize volumetric rendering code.

And of course water will have touch points with other systems:
- water interaction with terrain ... maybe you can paint a river on the terrain and that can be used to inform the river placement and geometry, as well as have that data fed to the material for splatting down river bottom terrain detail materials, or generate a falloff mask to splat sand or dirt on the river banks (so of this is in the terrain RFC design and is inherently already an aspect of the landscape canavs + gradient signals design.)
- Physicalized water (physx, collisions, buoyancy, etc,)
- water collisions driving events like: bullet/particle hits, ripples, FX entities (like splashes), audio, etc.

There are probably others not listed, this is just an general overview of the scope of work that generally comes along with water.

Ocean is another type of water, it probably has it's own concerns and specialized needs but there is likely some overlap.

**Comment by ghost** [comment]
Thank you guys. It is really awesome to see this as an ongoing development, especially for us who have lots of water-based type of game development (ship battles, combats, etc)

**Comment by ghost** [comment]
@AMZN-Liv @HogJonny-AMZN any news in regards to the infinite ocean/water gem and where in the pipeline it is?
